### PR TITLE
Clarify what happens if one activates a project without instantiating it

### PR DESCRIPTION
**Pull request by cossio**
I think the docs are not completely clear on what happens if one just `activate`s an environment but does not instantiate it. 
In this case Julia will throw an error if one attempts to `import` a package that is not already installed matching the precise version listed in the manifest file, if present.
This should be stated explicitly.

CC @IanButterworth

### CODE REVIEW

File: docs/src/environments.md [hunk 1]
```diff
@@ -105,6 +105,13 @@ Activating project at `~/Example.jl`
 If the project contains a manifest, this will install the packages in the same state that is given by that manifest.
 Otherwise, it will resolve the latest versions of the dependencies compatible with the project.
 
+Note that `activate` by itself does not install missing dependencies.
+In particular, if you attempt to import a package listed as a dependency in the project file, Julia will throw an error if this package is not already installed in the local depot.
+If the package is installed, but there is a manifest file, Julia will only load it if it finds an installed version matching the version listed in the manifest file.
+For this reason, `instantiate` is also needed in this example, since it will install all dependencies locally and in the versions required by the environment manifest file, if one is present.
+If the dependencies are already installed, `instantiate` does nothing.
```
Comment (IanButterworth) [inline docs/src/environments.md:112]:
```suggestion
If you only have a `Project.toml` a `Manifest.toml` must be generated by "resolving" the environment, then any misssing packages must be installed and precompiled. `instantiate` does all this for you.

If you already have a resolved `Manifest.toml`, then you will still need to ensure that the packages are installed etc. Again `instantiate` does this for you.

In short, `instantiate` is your friend to make sure an environment is ready to use. If there's nothing to do, `instantiate` does nothing.
```

File: docs/src/environments.md [hunk 2]
```diff
@@ -105,6 +105,14 @@ Activating project at `~/Example.jl`
 If the project contains a manifest, this will install the packages in the same state that is given by that manifest.
 Otherwise, it will resolve the latest versions of the dependencies compatible with the project.
 
+Note that `activate` by itself does not install missing dependencies.
+If you only have a `Project.toml` a `Manifest.toml` must be generated by "resolving" the environment, then any misssing packages must be installed and precompiled. `instantiate` does all this for you.
```
Comment (IanButterworth) [inline docs/src/environments.md:109]:
```suggestion
If you only have a `Project.toml`, a `Manifest.toml` must be generated by "resolving" the environment, then any missing packages must be installed and precompiled. `instantiate` does all this for you.
```

File: docs/src/environments.md [hunk 3]
```diff
@@ -105,6 +105,14 @@ Activating project at `~/Example.jl`
 If the project contains a manifest, this will install the packages in the same state that is given by that manifest.
 Otherwise, it will resolve the latest versions of the dependencies compatible with the project.
 
+Note that `activate` by itself does not install missing dependencies.
+If you only have a `Project.toml`, a `Manifest.toml` must be generated by "resolving" the environment, then any missing packages must be installed and precompiled. `instantiate` does all this for you.
+
+If you already have a resolved `Manifest.toml`, then you will still need to ensure that the packages are installed etc. Again `instantiate` does this for you.
```
Comment (cossio) [inline docs/src/environments.md:111]:
```suggestion
If you already have a resolved `Manifest.toml`, then you will still need to ensure that the packages are installed and with the correct versions. Again `instantiate` does this for you.
```

File: docs/src/environments.md [hunk 4]
```diff
@@ -105,6 +105,14 @@ Activating project at `~/Example.jl`
...
 Otherwise, it will resolve the latest versions of the dependencies compatible with the project.
 
+Note that `activate` by itself does not install missing dependencies.
+If you only have a `Project.toml`, a `Manifest.toml` must be generated by "resolving" the environment, then any missing packages must be installed and precompiled. `instantiate` does all this for you.
+
+If you already have a resolved `Manifest.toml`, then you will still need to ensure that the packages are installed and with the correct versions. Again `instantiate` does this for you.
+
+In short, `instantiate` is your friend to make sure an environment is ready to use. If there's nothing to do, `instantiate` does nothing.
+
+
```
Comment (IanButterworth) [inline docs/src/environments.md:115]:
```suggestion
```

**Review by KristofferC — APPROVED** [review]

**Comment by IanButterworth** [comment]
Hmm. MacOS segfaults?

**Comment by KristofferC** [comment]
> Hmm. MacOS segfaults?

Did not happen here: https://github.com/JuliaLang/Pkg.jl/runs/6749260642?check_suite_focus=true 

```
Julia Version 1.9.0-DEV.705
Commit b46c14e7316 (2022-06-02 14:49 UTC)
```

so something happened in the last 10 days I guess?

**Comment by IanButterworth** [comment]
MacOS segfaults are fixed, and this is just a static docs change so I'll just merge

**Pull request merged**